### PR TITLE
updates job status on queue inclusion

### PR DESCRIPTION
**Pull request by r0wdy1**

### CODE REVIEW

File: zp-relayer/pool/BasePool.ts [hunk 1]
```diff
@@ -44,9 +44,9 @@ export abstract class BasePool<N extends Network = Network> {
   abstract init(...args: any): Promise<void>
 
   abstract onSend(p: ProcessResult<any>, txHash: string): Promise<void>
-  abstract onConfirmed(p: ProcessResult<any>, txHash: string, callback?: () => Promise<void>): Promise<void>
+  abstract onConfirmed( p: ProcessResult<any>, txHash: string, callback?: () => Promise<void>,jobId?: string): Promise<void>
```
Comment (AllFi) [inline zp-relayer/pool/BasePool.ts]:
```suggestion
  abstract onConfirmed(p: ProcessResult<any>, txHash: string, callback?: () => Promise<void>, jobId?: string): Promise<void>
```

File: zp-relayer/pool/RelayPool.ts [hunk 2]
```diff
@@ -250,21 +250,58 @@ export class RelayPool extends BasePool<Network> {
...
+
+        await this.cacheTxLocally(res.commitIndex, res.outCommit, txHash, res.memo);
+      }
+    }
   }
 
-  async onConfirmed(res: ProcessResult<RelayPool>, txHash: string, callback?: () => Promise<void>): Promise<void> {}
+  async onFailed(txHash: string, jobId?: string): Promise<void> {
+    super.onFailed(txHash, jobId);
+    this.txStore.removeAll();
```
Comment (AllFi) [inline zp-relayer/pool/RelayPool.ts]: Shouldn't we remove only one index? If the previous transaction was confirmed not too long ago and is not indexed by the indexer yet, then we remove it as well if I understand correctly.

File: zp-relayer/pool/RelayPool.ts [hunk 3]
```diff
@@ -250,21 +250,58 @@ export class RelayPool extends BasePool<Network> {
...
+      if (!poolJob) {
+        logger.error('Pool job not found', { jobId });
+      } else {
+        poolJob.data.transaction.state = JobState.REVERTED;
+        poolJob.data.transaction.txHash = txHash;
+        await poolJob.update(poolJob.data);
+      }
+    }
+
+    // TODO: remove cached tx from txStore
```
Comment (AllFi) [inline zp-relayer/pool/RelayPool.ts]:
```suggestion
```

File: zp-relayer/workers/sentTxWorker.ts [hunk 4]
```diff
@@ -85,10 +85,10 @@ export async function createSentTxWorker({ redis, mutex, pool, txManager }: ISen
       // Successful
       jobLogger.info('Transaction was successfully mined', { txHash, blockNumber: tx.blockNumber })
 
-      await pool.onConfirmed(processResult, txHash, updatePoolJobState)
+      await pool.onConfirmed(processResult, txHash, updatePoolJobState, poolJobId)
     } else {
-      await pool.onFailed(txHash)
-      await updatePoolJobState()
+      await pool.onFailed(txHash, poolJobId);
+      //await updatePoolJobState()
```
Comment (AllFi) [inline zp-relayer/workers/sentTxWorker.ts]:
```suggestion
```

File: zp-relayer/pool/RelayPool.ts (hunk 2 above)
Comment (r0wdy1) [reply zp-relayer/pool/RelayPool.ts]: fixed by 0edb5e81b1ec5f45d8868825f48e1ac365844583

**Review by AllFi — APPROVED** [review]
LGTM

**Pull request merged**